src/shared/logger/logger.ts

```diff
@@ -52,22 +56,27 @@ export function compareLogLevel(l1: LogLevel, l2: LogLevel): number {
 
 /**
  * Gets the logger if it has been initialized
+ * @param type Gets the logger type:
+ * * `'main'` or `undefined`: Main logger; default impl: logs to log file and log output channel
+ * * `'channel'`: Channel Logger; default impl: logs to the `main` channels and the `AWS Toolkit` output channel
+ * * `'debug'`: Debug Console Logger; default impl: logs to the `channel` channels and the currently-active VS Code Debug Console pane.
```

Comment:
does debug logger not log to the log file? why? 

Why don't we have 1 logger and 1 channel? Why do we need more?

Reply:
The `debug` Logger is there solely to emulate the `emitMessage` functionality, which fills a similar niche to the channel logger but definitely should not have been there. re: your first comment, `emitMessage` specifically [didn't output to the logger](https://github.com/aws/aws-toolkit-vscode/pull/1451/files#diff-1a1cdd8c1e827ddfb4b3fe363a77fafe9cfdf53d812dfa78c10a9e00be046962L121), which really made me question why that was a construct in the channel logger in the first place. This also split the ANSI stripping, with the debug console retaining and the output channel dumping. Do we have reason to keep ANSI codes in any output channel/log file output? If not, we may want to add `removeAnsi` to those Winston transports to preserve this behavior.

re: one output + one channel, we've always split channels to keep noise to a minimum. Thinking we can discuss what channels and logs are necessary, but we may want to do that outside this PR; this is merely to migrate the functionality away from a separate impl.

Reply:
> `emitMessage` specifically [didn't output to the logger](https://github.com/aws/aws-toolkit-vscode/pull/1451/files#diff-1a1cdd8c1e827ddfb4b3fe363a77fafe9cfdf53d812dfa78c10a9e00be046962L121), which really made me question why that was a construct in the channel logger in the first place.

because emitMessage was added specifically to pass-through stuff to the channel, in places where the same data was already being logged to the logfile.

> Do we have reason to keep ANSI codes in any output channel/log file output? If not, we may want to add `removeAnsi` to those Winston transports to preserve this behavior.

removeAnsi is rarely needed, so the question is rather moot, and baking it into the logger is a one-way door.

> we've always split channels to keep noise to a minimum

I don't quite understand what you mean. The low-level "log file" is separate from the user-facing "AWS Toolkit" channel, yes. Other than that, no separation is needed.

Reply:
> because emitMessage was added specifically to pass-through stuff to the channel, in places where the same data was already being logged to the logfile.

Looking back, we basically have zero consistency on this...we have some instances where to verbose log to the logfile, and some times where we don't. Regardless, at its heart, `emitMessage` is a strange construct for this: don't output to logfile (like everything else the channelLogger did), and only strip ANSI from output channel messages because the debug console can show them. I'm fine with rethinking what we're doing here, but the fact of the matter is that this matches the functionality that we have before (minus the ANSI). It might be worth adding a flag when creating the output channel transport to strip ANSI in this case?

> I don't quite understand what you mean. The low-level "log file" is separate from the user-facing "AWS Toolkit" channel, yes. Other than that, no separation is needed.

I'm not sure what you're not sure about? The reason we have to have three different `getLogger` types is because of the above wonkiness. Either we figure out how to deal with the above or we have to have three different types. Juggling transports for a single logger is not an option; I played around with it in a `node` terminal, and adding/removing transports is extremely inconsistent in terms of what will be logged to what channels. If you want to try and juggle, I can try doing something a bit more scientific, but this seems to be the best option.

Reply:
we don't need to replicate the `emitMessage` functionality with this PR. `emitMessage` was only added because we had no way of emitting raw messages to a channel with the `TemplateHandler` thing.

Reply:
? We don't have any other way to emit to a debug console at this point. The fact of the matter is, this was built into the console logger class and is our only ingress point.

We can do away with this, but that means we do away with the `Debug Console` entirely.